plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1115,118 +1266,375 @@ def max_deflection(self):
+        thickness : `~astropy.units.Quantity`
+            The thickness of the layer, in units convertable to meters.
+
+        energy_axis : `~astropy.units.Quantity`
+            The energies corresponding to the stopping power array.
+
+        stopping_power : `~astropy.units.Quantity`
+            The stopping power in the material, multiplied by the material
+            mass density. The stopping power is tabulated in units of
+            MeV cm^2 / g, so this variable has units of MeV/cm.
+
+        active : boolean, optional
+            If True, this layer is marked as an active layer. The default is True.
+
+        name : str, optional
+            An optional name for the layer.
+
+        """
+        self.thickness = thickness
+        self.energy_axis = energy_axis
+        self.stopping_power = stopping_power
+        self.active = active
+        self.name = name
+
+
+class Stack:
+    r"""
+    A Stack is an ordered list of Layer objects.
+
+
+    Parameters
+    ----------
+
+    layers : list of `~plasmapy.diagnostic.charged_particle_radiography.Layer` objects
+        A list of the layers that make up the film stack.
+    """
+
+    def __init__(self, layers):
+        self._layers = layers
+        self._energy_bands = None
+
+    @property
+    def nlayers(self):
+        r"""
+        The number of layers in the stack.
+        """
 
-        ignore_grid: bool
-            If True, returns the intensity in the image plane in the absence
-            of simulated fields.
+        return len(self._layers)
 
-        optical_density: bool
-            If True, return the optical density rather than the intensity
+    @property
+    def nactive(self):
+        r"""
+        The number of layers in the stack marked 'active'
+        """
 
-            .. math::
-                OD = -log_{10}(Intensity/I_0)
+        return len([layer for layer in self._layers if layer.active])
 
-            where I_O is the intensity on the detector plane in the absence of
-            simulated fields. Default is False.
+    def deposition_curves(self, energies, return_only_active=True):
+        """
+        Calculates the deposition of an ensemble of particles over a range of
+        energies in a stack of films and filters.
+
+        Parameters
+        ----------
+
+        stack : list of Layer objects
+            This list of Layer objects defines the composition of the film stack.
+
+        energies : `~astropy.units.Quantity` array, shape [nenergies,]
+            Energies axis over which to calculate the deposition. Units convertable
+            to eV.
+
+        return_only_active : boolean, optional
+            If True, only the deposition in layers in which the `active` property
+            is True will be returned. This is usually desirable, since particles
+            captured in other layers will not be measured. If False, deposition in
+            all layers of the stack are returned. The default is True.
```

Comment:
```suggestion
        return_only_active : `bool`, optional
            If `True`, only the deposition in layers in which the `active` property
            is `True` will be returned. This is usually desirable, since particles
            captured in other layers will not be measured. If `False`, deposition in
            all layers of the stack are returned. The default is `True`.
```